src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -842,6 +925,113 @@ class CreateCloneStatementSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class WarehouseObjectPropertiesSegment(BaseSegment):
+    """A snowflake Warehouse Object Properties segment.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-warehouse.html
+    https://docs.snowflake.com/en/sql-reference/sql/alter-warehouse.html
+
+    Note: comments are handled seperately so not incorrectly marked as
+    warehouse object.
+    """
+
+    type = "warehouse_object_properties"
+
+    match_grammar = AnyNumberOf(
+        Sequence(
+            "WAREHOUSE_SIZE",
+            Ref("EqualsSegment"),
+            Ref("WarehouseSize"),
+        ),
+        Sequence(
+            "WAIT_FOR_COMPLETION",
+            Ref("EqualsSegment"),
+            OneOf("TRUE", "FALSE"),
```

Comment:
Nitpick: Is there an existing "boolean constant" segment? Wondering if we'd want to reuse that, e.g. so if we have something checking capitalization of boolean values, it would check this.

Reply:
Not a NitPick at all - great call! Updated.